### ISSUE TITLE
Unhandled promise rejection

### ISSUE THREAD

**Issue by devurandom** [openlink/structured-data-sniffer#11]
When watching a movie on Netflix, I see this on the browser console:

```
Unhandled promise rejection TypeError: promise.then is not a function
Stack trace:
wrapPromise/<@resource://gre/modules/ExtensionCommon.jsm:490:9
Promise@moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:9940:7
wrapPromise@resource://gre/modules/ExtensionCommon.jsm:489:14
callAsyncFunction@resource://gre/modules/ExtensionCommon.jsm:697:12
stub@resource://gre/modules/Schemas.jsm:2297:22
is_data_exist@moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/sniffer.js:366:13
@moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/sniffer.js:933:13
j@moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jquery-3.1.0.min.js:2:29566
Deferred/then/g/</k<@moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jquery-3.1.0.min.js:2:29882
 jsonld.js:9873:11
	onUnhandled/</result< moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:9873:11
	module.exports moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:4354:27
	onUnhandled/< moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:9867:16
	module.exports moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:10148:25
	<anonym> moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:4296:7
	run moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:4282:5
	listener moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6/lib/jsonld.js:4286:3
```

The copyright notice in `sniffer.js` suggests that `moz-extension://d13b1eab-e311-45ff-b72d-d871010a20d6` is the OpenLink Structured Data Sniffer.

**Comment by smalinin** [comment]
What kind of OSDS extension version did you use ?

**Comment by TallTed** [comment]
@devurandom - What browser (name, version, and host OS) are you running, and what version of the OSDS extension?

**Comment by devurandom** [comment]
Using Firefox 59.0.2 on Linux 4.16.6 with OSDS 2.16.6. It's reproducible on all pages, not specific to Netflix.

**Comment by smalinin** [comment]
Fixed, the problem is closed in new version 2.16.8.
